sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ScalaUDF.scala

```diff
@@ -1057,5 +1059,19 @@ case class ScalaUDF(
 
   private[this] val converter = CatalystTypeConverters.createToCatalystConverter(dataType)
 
-  override def eval(input: InternalRow): Any = converter(f(input))
+  val npeErrorMessage = "Given UDF throws NPE during execution, please check the UDF " +
+    "to make sure it handles null parameters correctly."
+
+  override def eval(input: InternalRow): Any = {
+    val result = try {
+      f(input)
+    } catch {
+      case e: NullPointerException =>
```

Comment:
1. For this code branch `eval(input: InternalRow)`, the existing NPE message should be clear enough if there is a full stacktrace, and  the stack contains method of the UDF.
2. The error message you provided can be totally wrong.
"Given UDF throws NPE during execution, please check the UDF to make sure it handles null parameters correctly". 

What if NPE is not caused by parameter? prompting this message is misleading.